Sources/ComposableArchitecture/Observation/Binding+Observation.swift

```diff
@@ -145,129 +118,110 @@
+    fileprivate let keyPath: KeyPath<State, Value>
 
-      public subscript<State: ObservableState, Action: ViewAction, Member: Equatable>(
-        dynamicMember keyPath: WritableKeyPath<State, Member>
-      ) -> Binding<Member>
-      where
-        Value == Store<State, Action>,
-        Action.ViewAction: BindableAction,
-        Action.ViewAction.State == State
-      {
-        Binding<Member>(
-          get: { self.wrappedValue.state[keyPath: keyPath] },
-          set: { newValue, transaction in
-            BindingLocal.$isActive.withValue(true) {
-              _ = self.wrappedValue.send(
-                .view(.binding(.set(keyPath, newValue))), transaction: transaction
-              )
-            }
-          }
-        )
-      }
+    public subscript<Member>(
+      dynamicMember keyPath: KeyPath<Value, Member>
+    ) -> _StoreBinding<State, Action, Member> {
+      _StoreBinding<State, Action, Member>(
+        binding: self.binding,
+        keyPath: self.keyPath.appending(path: keyPath)
+      )
     }
 
-    @available(iOS 17, macOS 14, tvOS 17, watchOS 10, *)
-    extension SwiftUI.Bindable {
-      public subscript<State: ObservableState, Action: BindableAction, Member: Equatable>(
-        dynamicMember keyPath: WritableKeyPath<State, Member>
-      ) -> Binding<Member>
-      where Value == Store<State, Action>, Action.State == State {
-        Binding<Member>(
-          get: { self.wrappedValue.state[keyPath: keyPath] },
-          set: { newValue, transaction in
-            BindingLocal.$isActive.withValue(true) {
-              _ = self.wrappedValue.send(
-                .binding(.set(keyPath, newValue)), transaction: transaction
-              )
-            }
-          }
-        )
-      }
+    /// Creates a binding to the value by sending new values through the given action.
+    ///
+    /// - Parameter action: An action for the binding to send values through.
+    /// - Returns: A binding.
+    public func sending(_ action: CaseKeyPath<Action, Value>) -> Binding<Value> {
+      self.binding[state: self.keyPath, action: action]
+    }
+  }
 
-      public subscript<State: ObservableState, Action: ViewAction, Member: Equatable>(
-        dynamicMember keyPath: WritableKeyPath<State, Member>
-      ) -> Binding<Member>
-      where
-        Value == Store<State, Action>,
-        Action.ViewAction: BindableAction,
-        Action.ViewAction.State == State
-      {
-        Binding<Member>(
-          get: { self.wrappedValue.state[keyPath: keyPath] },
-          set: { newValue, transaction in
-            BindingLocal.$isActive.withValue(true) {
-              _ = self.wrappedValue.send(
-                .view(.binding(.set(keyPath, newValue))), transaction: transaction
-              )
-            }
-          }
-        )
-      }
+  @available(iOS 17, macOS 14, tvOS 17, watchOS 10, *)
+  @dynamicMemberLookup
+  public struct _StoreBindable_SwiftUI<State: ObservableState, Action, Value> {
```

Comment:
While I think it might be possible to abstract these 3 "store binding" types, I'm not sure it's worth it or a good idea.